dlnd/dlnd_datamode_rof_forcing_mod.F90

```diff
@@ -107,105 +115,139 @@ end subroutine dlnd_datamode_rof_forcing_advertise
-      do n = 1,size(fldnames_h2o)
-         fieldname = trim(fldnames_h2o(n))
-         call dshr_dfield_add( dfields, sdat, trim(fieldname), trim(fieldname), exportState, logunit, mainproc, rc)
-         if (chkerr(rc,__LINE__,u_FILE_u)) return
-      end do
+      call shr_strdata_get_stream_pointer(sdat, 'Flrl_rofsur' , strm_Flrl_rofsur, requirePointer=.true., &
+           errmsg=trim(subname)//'ERROR: strm_Flrl_rofsur be associated for dlnd rof_forcing mode', rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+      call shr_strdata_get_stream_pointer(sdat, 'Flrl_rofsub' , strm_Flrl_rofsub, requirePointer=.true., &
+           errmsg=trim(subname)//'ERROR: strm_Flrl_rofsub be associated for dlnd rof_forcing mode', rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+      call shr_strdata_get_stream_pointer(sdat, 'Flrl_rofgwl' , strm_Flrl_rofgwl, requirePointer=.true., &
+           errmsg=trim(subname)//'ERROR: strm_Flrl_rofgwl be associated for dlnd rof_forcing mode', rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+      call shr_strdata_get_stream_pointer(sdat, 'Flrl_rofi' , strm_Flrl_rofi, requirePointer=.true., &
+           errmsg=trim(subname)//'ERROR: strm_Flrl_rofi be associated for dlnd rof_forcing mode', rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+      ! optional stream field pointer
+      call shr_strdata_get_stream_pointer(sdat, 'Flrl_irrig' , strm_Flrl_irrig, rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
    end subroutine dlnd_datamode_rof_forcing_init_pointers
 
    !===============================================================================
-   subroutine dlnd_datamode_rof_forcing_advance(exportState, rc)
-
-      ! input/output variables
-      type(ESMF_State)      , intent(inout) :: exportState
-      integer               , intent(out)   :: rc
+   subroutine dlnd_datamode_rof_forcing_advance()
 
       ! local variables
-      integer           :: n,nfld
-      real(r8), pointer :: fldptr1(:)
-      real(r8), pointer :: fldptr2(:,:)
-      character(CS)     :: fieldname
+      integer :: ni,nf
       character(len=*), parameter :: subname='(dlnd_datamode_rof_forcing_advance): '
       !-------------------------------------------------------------------------------
 
-      rc = ESMF_SUCCESS
-
-      if (ntracers_nonh2o > 0) then
-         ! Set special value over masked points
-         if (ntracers_nonh2o == 1) then
-            call dshr_state_getfldptr(exportState, Flrl_rofsur_nonh2o, fldptr1=fldptr1, rc=rc)
-            if (chkerr(rc,__LINE__,u_FILE_u)) return
-            do n = 1,size(fldptr1)
-               if (lfrac(n) == 0._r8) fldptr1(n) = 1.e30_r8
+      if (ntracers_nonh2o > 1) then
+         ! Note that the inner dimension is the field index
+         do nf = 1,ntracers_nonh2o
+            do ni = 1,size(Flrl_rofsur_nonh2o_2d,dim=2)
+               if (lfrac(ni) == 0._r8) then
+                  Flrl_rofsur_nonh2o_2d(nf,ni) = 1.e30_r8
```

Comment:
Is `1.e30_r8` a special value (maybe `SHR_CONST_SPVAL`)? Please use a parameter for that repeated naked constant in this module.

Reply:
done.